src/QueryResults.ts

```diff
@@ -0,0 +1,61 @@
+import Promise from 'dojo-core/Promise';
+import Task from 'dojo-core/async/Task';
+import {duplicate, mixin} from 'dojo-core/lang';
+import * as dstore from './interfaces';
+
+export interface QueryOptions {
+	totalLength?: number| Promise<number>;
+	response: any;
+	length?: number;
+	forEach?: <T>(callback:(value: T, index: number, source: T[]) => any, instance: any) => Promise<void>;
+}
+
+function forEach<T>(callback: (value: T, index?: number, array?: T[]) => void, thisObject?: any): Promise<void> {
+	const execute = function (resolve: any, reject: any) {
```

Comment:
I think `resolve` should be of type `(value?: T | Promise.Thenable<T>) => void` and `reject` should be of type `(reason?: any) => void` (from https://github.com/dojo/core/blob/master/src/Promise.ts#L26). The current definition would allow for you to pass anything in.